src/main/java/discord4j/discordjson/json/ApplicationInfoRequest.java

```diff
@@ -0,0 +1,44 @@
+package discord4j.discordjson.json;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import discord4j.discordjson.possible.Possible;
+import org.immutables.value.Value;
+
+import java.util.List;
+
+@Value.Immutable
+@JsonSerialize(as = ImmutableApplicationInfoRequest.class)
+@JsonDeserialize(as = ImmutableApplicationInfoRequest.class)
+public interface ApplicationInfoRequest {
+
+    static ImmutableApplicationInfoRequest.Builder builder() {
+        return ImmutableApplicationInfoRequest.builder();
+    }
+
+    @JsonProperty("custom_install_url")
+    Possible<String> customInstallUrl();
+
+    @JsonProperty("description")
+    Possible<String> description();
+
+    @JsonProperty("role_connections_verification_url")
+    Possible<String> roleConnectionsVerificationUrl();
+
+    @JsonProperty("install_params")
+    Possible<InstallParamsData> installParams();
+
+    Possible<Integer> flags();
+
+    Possible<byte[]> icon();
+
+    @JsonProperty("cover_image")
+    Possible<byte[]> coverImage();
```

Comment:
To edit image data we should update `Possible<byte[]>` to `Possible<Optional<String>>` to be consistent with how it's done at GuildModifyRequest:

https://github.com/Discord4J/discord-json/blob/969e63be7630fbbb58ed4c0ed34fe5b0c1f9e23f/src/main/java/discord4j/discordjson/json/GuildModifyRequest.java#L40-L50

Wrapping in Optional is required to model clearing out the image data (sending `null` to the API) [docs](https://discord.com/developers/docs/resources/guild#modify-guild)

In Discord4J we can later create a ApplicationEditSpecGenerator similar to GuildEditSpecGenerator by passing an `Image` object.

https://github.com/Discord4J/Discord4J/blob/6cf8ff2263cd204c1e0e3d9e276f0db91ac85ab2/core/src/main/java/discord4j/core/spec/GuildEditSpecGenerator.java#L55-L63

Reply:
Updated in https://github.com/Discord4J/discord-json/pull/158/commits/97853f61c577691d0d883400f8076626b444b93b

Also updated https://github.com/Discord4J/Discord4J/pull/1164 to use the spec generator just like in guilds